src/compile/data/joinaggregate.ts

```diff
@@ -26,7 +26,9 @@ export class JoinAggregateTransformNode extends DataFlowNode {
   public dependentFields() {
     const out = new Set<string>();
 
-    this.transform.groupby.forEach(f => out.add(f));
```

Comment:
Instead you can write (foo || []).forEach or even better also use a for of loop.